### PR TITLE
Rebalance archery take... 5?

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: Balance "Overhaul archery balance."

#### Purpose of change
As outlined in #35971, #30681 and #26183 and started in #28069, archery balance has been up in the air for a very long time now.
After addressing effective range in #37199 I'm taking a look at a pile of other effectiveness mechanisms here.
The goal is making archery in general be more representative of reality, and in broad strokes that means applying the planning outlined in #26183 and https://docs.google.com/spreadsheets/d/1wpuAk7F1fJ227yAgYbokwn7KxdEeOv-MGFvnnnW5s1Q/edit#gid=781207725 to normalize archery damage potential while shifting a significant part of the damage contribution to multipliers on arrowheads instead of having the damage be additive.

#### Describe the solution
- [x] Adjust arrow damage multipliers.
- [x] Squashed reload times.
- [x] Calculate momentum for bows and crossbows.
- [x] Map momentum figures to damage.
- [x] Tweak damage for interaction with zombies.
- [x] Retune draw weight figures for bows.

The rationale for the damage numbers is that we make some assumptions about weapon draw weight and efficiency based on the understanding of what real-world item each weapon represents, and once we have an estimated power output, we take the square root of the presented projectile momentum as the base damage number. Then this number is modified by the terminal ballistics characteristics of the arrow, mostly the head, to determine the final damage number.  In practice this falls into a handful of categories.
0.5 multiplier from blunt-tip arrows used for hunting small game.
0.75 for poorly sharpened tips, such as a bare wood tip.
1.0 for a decently sharp spike of a tip, exemplified by a bodkin tip used to maximize armor piercing.
1.25 for a crude bladed tip.
1.5 for a modern broadhead.
1.75 for a particularly high performance arrow with an advanced modern broadhead tip.
There might be some ambiguity about which arrow falls into which category, so I could readily see issues with the assignment of these values.

A somewhat more arbitrary number is the mapping of strength stat to weapon poundage.
The document that Darktoes put together perpetuates the assumption that each point of strength stat is good for 10 pounds of draw weight on a bow.  This doesn't make much sense, as it places the low end of bows at 30lbs within the reach of an incredibly feeble str 3 character, and the main "used by fit hunters" 60lb draw bows in reach of a still significantly-weaker-than-average str 6 character.  AFAICT the goal here was to keep the mapping trivial and to have that simple mapping span all the way to nonsensical 200lb draw weight bows.  I've instead opted to try and place the 30lb draw at the low end of strength but not utterly feeble, and to place the 60 lb draw at a bit over average, and then the very high draw weight bows land in the upper limits of strength.

There's no coherent case for crossbow poundage limits, so these are all set at 8 when the style of crossbow calls for it, and the limit is absent in cases when there is some built-in mechanical advantage.

A somewhat odd issue is that the reload time for various bows was all over the place, I've squashed them all to 30 moves as I have a great deal of trouble justifying a second or more to place an arrow on a string.

##### Damage mapping
Here's a set of break points for hunting I've been able to find
All arrows weigh 30g / 463 grains

FPS | ft-lbs | slug-ft-s | Game
--|--|--|--
136 | 19.01 | 0.28 | Turkeys
153 | 24 | 0.315 | Whitetails at close range
195 | 39 | 0.401 | Whitetails
249 | 63.73 | 0.512 | Elk
278 | 79.44 | 0.572 | Moose

#### Describe alternatives you've considered
Compared to the previous PR, I removed the majority of string changes considering that the 0.E release is just around the corner, adjusted the strength requirements, refrained from doubling damage resulting from the kinetic energy analysis, and refrained from adjusting incidental values such as the range stat.

### CODE REVIEW

**Comment by 1n17** [comment]
This is not right, if going as base bow numbers as kinetic energy then blunt tip needs to deliver 'bullet equivalent' damage.
Eg. longbow fired arrow should come close to .22 bullet at effective range - when pure energy is considered, but due to it having cutting arrowhead the wounding caused will be vastly greater than that of .22

**Comment by Saint-of-Grey** [comment]
I wouldn't assume bullets damage the same way arrows do, but easily digested data is a tad difficult to track down. A quick search shows this has been a hotly debated subject between bow hunters and gun hunters for quite some time, with plenty of dickwaving on both sides.

The only real direct comparison I can track down is bored people shooting [buckets full of sand](https://www.youtube.com/watch?v=hDhkdKK3vCg) or [even body armor](https://www.youtube.com/watch?v=DfT928MEwZk) using both and the arrow has far more penetration power than the bullet in these cases. As it would turn out, this isn't just something you have to wonder about, as someone has written [a rather extensive physics paper on the subject of arrow terminal ballistics](https://cpw.state.co.us/Documents/Hunting/EHU/Momentum-KineticEnergy-ArrowPenetration.pdf). Some images I tracked down during this search showed that the ballistic profiles both have a fair amount of depth: ([link](https://www.luckygunner.com/labs/self-defense-ammo-ballistic-tests/) [link](https://bowhunting.net/2011/09/broadhead-tests-and-ballistic-gelatin/) [link](https://www.researchgate.net/figure/The-ballistics-gel-target-with-embedded-arrows-a-target-draped-with-caribou-hide-with_fig1_253697879)).

**Comment by 8street** [comment]
I can also add that in real life arrows have broadhead tips and sport tips. Broadheads have more damage, but sports penetrate much better in my opinion. 

Also, the range of all bows is underestimated. The aiming range is much smaller and corresponds to the max game range now. I can shoot at a distance of 100m from my recurve 30 pound bow. But precisely I can shoot only at 20m and smaller (in a target about the size of a man). And I'm an ordinary amateur.

**Comment by mlangsdorf** [comment]
As an aside, I'm extremely dubious about the rapid rate of loosing for Mongol bows.  For all bows, kinetic energy of the arrow going out has to be less than the kinetic energy the archer put into bending the bow, and the bow just lets you store KE slowly and release it quickly.  The faster you're loosing arrows, the less time you have to slowly store KE and the less advantage the bow gives you over just throwing arrows at people.  If you're loosing 3 arrows in 1.5 seconds, they're going to have a lot less KE than the arrow from the guy who loosed a single arrow in 3 seconds.

**Comment by kevingranade** [comment]
> blunt tip needs to deliver 'bullet equivalent' damage.

Blunt tip is not a bullet like tip, it is specifically a very wide tip designed to not penetrate into the target.

A "field point" is a bullet like tip, and unless I missed it, those get the same multiplier as bullets do.

> some power focused constructions like compound bow tend to take longer to draw than eg. mongol recurve that good archer could supposedly shoot 2-3 times a second.

> As an aside, I'm extremely dubious about the rapid rate of loosing for Mongol bows.

"reload" is not "draw time", it is "time to place the arrow on the string".

It might be illustrative to break down the sequence of events
1. Open firing menu with bow equipped (no time spent)
2. Select an arrow to fire (obtain cost for inventory retrieval)
3. Reload time for bow (the time we're talking about, *just* placing an arrow in the hand onto the string)
4. Aim time (varies wildly, from 0 moves to multiple turns)
5. Firing/release time (I just dropped this in #37230 to one second or less, all the way down to .2 sec for an expert archer)

4 almost always dominates total time, when it doesn't you're very unlikely to hit anything.
5 is only interesting in that it might end your turn and give an approaching monster a chance to act before you get another turn.
We're implicitly assuming that "time to draw" is baked into 4 and 5, and I'm quite confident that for the use cases we care about, those are sufficient to cover any draw time.

> I can also add that in real life arrows have broadhead tips and sport tips. Broadheads have more damage, but sports penetrate much better in my opinion.

This is already accounted for, different arrows have either a high damage multiplier, or some amount of bonus armor penetration.  Ideally this would be some kind of zero sum sliding scale where you can trade damage for penetration and vice versa, but we have it manually specified in the bow and arrow definitions.

> Also, the range of all bows is underestimated.

This is out of scope for this issue, but I'll just say that effective range is scaled the same way gun ranges are, edge of visible area is not treated as 60 meters away, more like 500 meters away.

> User will not be able to stretch properly such a strong bow.

A typical person will not, but DDA characters are not restricted to typical levels of strength.

> I think it would help if you added more comments defining the bows - overall size, material, draw length if you have them, but delivered arrow energy is probably most important.

I'll add the pertinent assumptions from the spreadsheet to the bow and arrow entries as comments.

> I found SoG's article on arrow terminal ballistics to be fairly persuasive

UGGGGHHHHH, yea me too, I'll take a look at some figures and see where it pans out.

**Comment by kevingranade** [comment]
OK I've convinced myself we can switch to a momentum-based model without too much trouble.
Force expended by the bow is fixed.
That force goes into accelerating the bow arms and string and the arrow.
Part of efficiency is the ratio of bow limb mass vs arrow mass. Whichever is heavier gets more force applied to it.
Bow limbs and string are not a simple fixed mass due to rotational inertia, but can be treated as such.

So what we have is:
P. Bow Power (draw weight * some goofy constant * draw length).
Ml. Bow limb mass (not actually mass, but representative of it).
Ma. Arrow mass (this is actually just mass).
Ah. Arrowhead mechanical advantage.

The formula for arrow damage is something like:
D = f( P / Ml ) * g( Ma * Ah )

f(P * Ml) is encoded as bow "ranged_damage".
g(Ma * Ah) is encoded as arrow "prop_damage"

f() is something similar to sqrt() to handle scaling as more power is dumped into the system.
g() is potentially something similar, or it might just be the identity function.

That's all for plugging a bunch of numbers in, at the end of the day we can still just multiply the two terms to get expected damage.

Caveats:
Heavier arrows should have a better efficiency, but also have an accuracy penalty since they're moving slower and have a more curved trajectory.  Higher power bows should possibly have an accuracy bonus to account for their higher speed and therefore flatter trajectories.

**Comment by kevingranade** [comment]
After fiddling around with the formulas a bit, I realized that we can't split out bow limb weight and arrow weight, they are part of the same interrelated evaluation.  What we CAN do however is standardize the mass of our arrows to eliminate this discrepancy, and once we factor these pieces back out we can re-add different masses of arrows and different effects on arrow selection.

**Comment by 1n17** [comment]
> As an aside, I'm extremely dubious about the rapid rate of loosing for Mongol bows.

Don't quote me on that, just distinctly remember that bit about about mongol or some other asian recurves when doing my own research once the whole bow debacle started. I'm guessing they counted arrow that was still being drawn as shoot for it to make sense though, since from my practical tests of fast shooting techniques I was unable to quite reproduce even 1.5 releases/s - it was a rather heavy (for me being a novice at least), ~45lbs borrowed recurve though.

**Comment by mlangsdorf** [comment]
This is one of the reasons I want to know delivered energy of the arrow, because if you know arrow kinetic energy KE, you can do:
KE = 0.5 * m * v^2
v = sqrt( 2 * KE / m )
momentum = m * ( 2 * KE / m )^0.5

and as I have mentioned before, Doug Cole has worked out extremely complicated formulas for determining the KE of the arrow based on the bow design, thickness, length, draw length, and materials.  Which basically work out to 45J on the low end to 200-300J on the high end and we can assign numbers that look right and match people's expectations of bow performance without getting into the nitty gritty of bow design.

**Comment by kevingranade** [comment]
Just an update, I'm still plugging away at this, but some of the momentum numbers are fairly hard to come by, and the stats in the spreadsheet from The Lethal Spring are incoherent (0.3 lb arrows?!?). I actually tried to buy The Lethal Spring, but couldn't get SHG to take any of my payment options.
I'm filling in the numbers, but it's slow going for various reasons.

**Comment by kevingranade** [comment]
Progress update, obviously I'm still poking at this.  I managed to get my copy of deadly spring, and it seems the cribbed stats I was basing things on were critically flawed in that someone transcribed arrow diameter in inches as arrow weight in lbs.  Well that's obviously not correct, it's not even off by some factor it's just wrong.  Now that I have the real numbers (which I've put in the item definitions as comments) it became MUCH easier to determine momentum and also place those in those comments, so now we have all the bows on a linear scale based on momentum.

The range is 0.2 or so to 0.8 ft-lb-s for "normal" bows, but the "huge compound bow" is at 1.4 and the "Rantz-Gustang" I've modelled as a 1,000 lb draw arbalest with steel arms, which puts it at 4, which is crazy high but not totally unreasonable.

The next step is mapping momentum to damage, here's the plan for that.
Take recommended bow loadouts for hunting various game and build a table of "this bow should be able to take down a X with a good shot, but not Y".
Encode the above table as a unit test that scores good hits on the specified targets with the specified bows and then asserts whether the target is killed or not.
Fiddle with a linear conversion rate that makes the tests pass.
Sanity check the results.
In particular sanity check that the results vs zombies are what we want them to be, because a bow shot that will kill a deer by making it bleed out is not going to work the same on a zombie.

**Comment by kevingranade** [comment]
I'm still looking at balancing damage vs animals against damage vs zombies, pursuing a few ideas:
1. Make zombies no longer vulnerable to criticals and headshots. This means the rather large (1.5x to 2.0) boost to arrow damage that makes them one-hit natural animals doesn't apply to zombies.  This has the side effect of nerfing all ranged damage vs zombies, which I'm pretty much ok with, but this might be a too-large change for this point in development.
2. Add a "immediately fatal wound" mechanic where if you deal e.g. 50% of a target's HP in a single hit as a crit, it escalates to killing the target outright.

**Comment by ashGlaw** [comment]
I prefer the first one. That idea seems very straightforward, and it should make zombies a bit more threatening. 

The second one could be cool, but the particulars are make or break on a feature like that. I think modeling shock and bloodloss from injury and all that would be rad, but nobody likes to get one shotted, and I fear that mechanic could be unnecessarily frustrating for players. Then again, it could be great, the issue is one of complexity; which is a great argument in favor the first idea

**Comment by kevingranade** [comment]
Riffing on the second one, another option is to hide critical hits and headshots behind a per enemy damage threshold, so you only get the damage multiplier if you get the positioning of a critical hit, plus dealing enough damage to trigger it.

**Comment by Zireael07** [comment]
That per-enemy dmg threshold for crits is a good idea, imho.

**Comment by kevingranade** [comment]
Unless I'm missing something all I have left is auditing bow draw weights.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/so-how-does-the-new-archery-work/19850/37

File: tests/archery_damage_test.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+#include "game_constants.h"
+#include "item.h"
+#include "monster.h"
+#include "projectile.h"
+
+static void test_projectile_attack( std::string target_type, bool killable,
+                                    dealt_projectile_attack &attack, std::string weapon_type )
+{
+    for( int i = 0; i < 10; ++i ) {
+        monster target{ mtype_id( target_type ), { 0, 0, 0 } };
```
Comment (ifreund) [inline tests/archery_damage_test.cpp]: clang-tidy doesn't like this
```suggestion
        monster target{ mtype_id( target_type ), tripoint_zero };
```

File: tests/archery_damage_test.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    item weapon( weapon_type );
+    // The standard modern hunting arrow, make this a parameter if we extend to crossbows.
+    weapon.ammo_set( ammo_type, 1 );
+
+    projectile test_projectile;
+    test_projectile.speed = 1000;
+    test_projectile.impact = weapon.gun_damage();
+
+    dealt_projectile_attack attack {
+        test_projectile, nullptr, dealt_damage_instance(), { 0, 0, 0 }, accuracy_critical - 0.05
```
Comment (ifreund) [inline tests/archery_damage_test.cpp]:
```suggestion
        test_projectile, nullptr, dealt_damage_instance(), tripoint_zero, accuracy_critical - 0.05
```

File: doc/GAME_BALANCE.md [hunk 3]
```diff
@@ -186,7 +186,7 @@ Pawn shop         | Anything except milspec weighted towards unusual calibers |
...
 
 ## Ammo stats
-The damage (**Dmg**) of ammunition is the square root of a round's muzzle energy in joules (**Energy, J**) rounded to the nearest integer with an arbitrary increase or decrease to account for terminal ballistics. Damage of handloaded ammo is set to 92% (rounded down) of their factory counterparts. A similar system for calculating recoil is planned but not currently being worked on. The figures used to calculate stats and any other relevant information are presented in table below.
+The damage (**Dmg**) of ammunition is the square root of a round's muzzle energy in joules (**Energy, J**) rounded to the nearest integer with an arbitrary increase or decrease to account for terminal ballistics. Damage of handloaded ammo is set to 92% (rounded down) of their factory counterparts. A similar system for calculating recoil is planned but not currently being worked on. The figures used to calculate stats and any other relevant information are presented in table below.  Bows and arrows use a similar system, but energy is defined in the bow or crossbow, and the arrow provides an adjustment for terminal ballistics with the prop_damage field.
```
[the remark this answers is not in the document]
Comment (kevingranade) [reply doc/GAME_BALANCE.md]: Reminder for me, I still need to revisit this and outline how it's intended to work now.

File: data/json/items/crossbows.json [hunk 4]
```diff
@@ -416,34 +415,24 @@
...
+    "flags": [ "FIRE_TWOHAND", "PRIMITIVE_RANGED_WEAPON", "TRADER_AVOID" ],
     "skill": "rifle",
+    "min_strength": 8,
     "ammo": "bolt",
     "weight": "6803 g",
     "volume": "5 L",
     "bashing": 20,
     "to_hit": -1,
-    "ranged_damage": 50,
+    "ranged_damage": 22,
```
[the remark this answers is not in the document]
Comment (yarbelk) [reply data/json/items/crossbows.json]: reasoning: medieval crossbows had _very_ short draw lengths due to limits in material science and the reasonable danger of 1000lb tensions right next to your face.

**Comment by I-am-Erk** [comment]
Is there anything left waiting on this before merge?

**Comment by kevingranade** [comment]
It's ready to go.

**Comment by anothersimulacrum** [comment]
There are merge conflicts.

**Comment by kevingranade** [comment]
Conflicts resolved.

**Pull request merged**